ledger/acctupdates.go

```diff
@@ -1229,11 +1229,16 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 			treeTargetRound = dbRound + basics.Round(offset)
 		}
 		for i := uint64(0); i < offset; i++ {
-			err = accountsNewRound(tx, deltas[i], creatableDeltas[i], roundTotals[i+1].RewardsLevel, protos[i+1])
+			err = accountsNewRound(tx, deltas[i], creatableDeltas[i])
 			if err != nil {
 				return err
 			}
 		}
+		err = totalsNewRounds(tx, deltas[:offset], roundTotals[1:offset+1], protos[1:offset+1])
```

Comment:
This might not be possible, but could `roundTotals` or `protos` ever have fewer than 1 entry? For example, if `offset` is 0, the above loop will never be called. So this refactor does change the semantics of this block of code slightly.

Reply:
I think that having offset of 0 is not a possible scenario here. It's protected by the enqueueing function `committedUpTo` by doing:
```
...
	defer func() {
		au.accountsMu.RUnlock()
		if dc.offset != 0 {
			au.committedOffset <- dc
		}
	}()
...
```

The `roundTotals` as well as the `protos` are always having one more element compared to the `deltas`. The preamble retains that property:
```
	copy(deltas, au.deltas[:offset])
	copy(creatableDeltas, au.creatableDeltas[:offset])
	copy(roundTotals, au.roundTotals[:offset+1])
	copy(protos, au.protos[:offset+1])
```

The `offset -= uint64(au.dbRound - dbRound)` would never bring the `offset` to zero as long as the blocks numbers are growing.